clients/js/src/embeddings/CohereEmbeddingFunction.ts

```diff
@@ -5,26 +5,56 @@ let CohereAiApi: any;
 export class CohereEmbeddingFunction implements IEmbeddingFunction {
     private api_key: string;
     private model: string;
+    private cohereAiApi?: any;
 
     constructor({ cohere_api_key, model }: { cohere_api_key: string, model?: string }) {
-        try {
-            // eslint-disable-next-line global-require,import/no-extraneous-dependencies
-            CohereAiApi = require("cohere-ai");
-            CohereAiApi.init(cohere_api_key);
-        } catch {
-            throw new Error(
-                "Please install the cohere-ai package to use the CohereEmbeddingFunction, `npm install -S cohere-ai`"
-            );
-        }
+        // we used to construct the client here, but we need to async import the types
+        // for the openai npm package, and the constructor can not be async
         this.api_key = cohere_api_key;
         this.model = model || "large";
     }
 
     public async generate(texts: string[]) {
-        const response = await CohereAiApi.embed({
+
+        if (this.cohereAiApi === undefined) {
```

Comment:
I would move this part into the import function. Because it is import logic and not generation logic. It would make the generate function cleaner if you would do

```js
await CohereEmbeddingFunction.import();
const response = await this.cohereAiApi.embed({
```

The import function would then immediately return if the package is already imported.

Reply:
you could move it to another function called "loadClient"
something like
```js
private async loadClient(){

if(this.cohereAiApi) return;
   try {
                // eslint-disable-next-line global-require,import/no-extraneous-dependencies
                const { cohere } = await CohereEmbeddingFunction.import();
                CohereAiApi = cohere;
                CohereAiApi.init(this.api_key);
            } catch (_a) {
                // @ts-ignore
                if (_a.code === 'MODULE_NOT_FOUND') {
                    throw new Error("Please install the cohere-ai package to use the CohereEmbeddingFunction, `npm install -S cohere-ai`");
                }
                throw _a; // Re-throw other errors
            }
            this.cohereAiApi = CohereAiApi;

        }


}
```

Reply:
i like this change - thanks! will implement